ddtrace/appsec/trace_utils.py

```diff
@@ -121,3 +125,52 @@ def track_custom_event(tracer, event_name, metadata):
     for k, v in six.iteritems(metadata):
         span.set_tag_str("%s.%s.%s" % (APPSEC.CUSTOM_EVENT_PREFIX, event_name, k), str(v))
         span.set_tag_str(constants.MANUAL_KEEP_KEY, "true")
+
+
+def should_block_user(tracer, userid):  # type: (Tracer, str) -> bool
+    """
+    Return true if the specified User ID should be blocked.
+
+    :param tracer: tracer instance to use
+    :param userid: the ID of the user as registered by `set_user`
+    """
+
+    if not config._appsec_enabled:
+        log.warning("should_block_user call requires ASM to be enabled")
+        return False
+
+    # Early check to avoid calling the WAF if the request is already blocked
+    span = tracer.current_root_span()
+    if _context.get_item(WAF_CONTEXT_NAMES.BLOCKED, span=span):
+        return True
+
+    _asm_request_context.call_waf_callback(custom_data={"REQUEST_USER_ID": str(userid)})
+    return bool(_context.get_item(WAF_CONTEXT_NAMES.BLOCKED, span=span))
+
+
+def block_request():  # type: () -> None
+    """
+    Block the current request and return a 403 Unauthorized response. If the response
+    has already been started to be sent this could not work.
+    """
+    if not config._appsec_enabled:
+        log.warning("should_block_user call requires ASM to be enabled")
```

Comment:
Update warning to explicitly state the feature is disabled

Reply:
Done (didn't refer explicitly to the 1-click feature since this could be used by the user to block any request, just need ASM enabled).